src/ott/neural/data/dataloaders.py

```diff
@@ -0,0 +1,133 @@
+# Copyright OTT-JAX
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict, Iterator, Mapping, Optional
+
+import numpy as np
+
+__all__ = ["OTDataLoader", "ConditionalDataLoader"]
+
+
+class OTDataLoader:
+  """Data loader for OT problems.
+
+  Args:
+    batch_size: Number of samples per batch.
+    source_lin: Linear part of the source measure.
+    source_quad: Quadratic part of the source measure.
+    target_lin: Linear part of the target measure.
+    target_quad: Quadratic part of the target measure.
+    source_conditions: Conditions of the source measure.
```

Comment:
condition is maybe a bit vague? I think this could benefit from explaining. Can we also add some ``[batch, d]`` exemplar sizes? is the conditions a batched object (à la GENOT), or is it the same for all the batch? (à la CondOT)

Reply:
This depends on the dataloader. If we use the `OTDataLoader` it's not the same one for all samples. An example usage would be conditional generative modeling, where the conditions correspond to text embeddings. If we use the `ConditionalDataLoader`, it's the same condition per batch. Btw, the conditions of GENOT are "different conditions". They are constructed from the data. Yet, we can add (even more) conditions to GENOT, by providing conditions here, i.e. 1) either providing conditions like in generative modeling or 2) extending GENOT to the CondGENOT setting. The former should be done with the `OTDataLoader`, the latter with the `ConditionalDataLoader`. 

I know it's a little bit complicated, but I think this way it's most generic. What do you think @marcocuturi @michalk8 ? Where should we explain this?